stdlib/asm/collections/mmr.masm

```diff
@@ -1,6 +1,75 @@
 use.std::mem
 use.std::crypto::hashes::native
 
+#! Computes leading number of ones in `number`.
+#!
+#! Stack transition:
+#! Input: [number, ...]
+#! Output: [leading_ones, ...]
+#! Cycles: 6 + 11 * leading_ones
+export.u32unchecked_leading_ones
+  # push counter and flag, the flag is true while no zero have been seen (2 cycles)
+  push.0 push.1
+  # => [flag, count, number, ...]
+
+  dup
+  # LOOP: [flag, count, number, ...] (1 cycles)
+
+  while.true
+    # update the flag (2 cycles)
+    dup.2 u32checked_and
+    # => [flag, count, number, ...]
+
+    # update the number (4 cycles)
+    movup.2 u32unchecked_div.2
+    # => [number/2, flag, count, ...]
+
+    # update the counter (3 cycles)
+    movup.2 dup.2 add
+    # => [count, number/2, flag, ...]
+
+    # check if should continue (2 cycles)
+    movup.2 dup
+    # LOOP: [flag, flag, count, number/2, ...]
+  end
+
+  # drop flag and number (3 cycles)
+  drop swap drop
+  # => [count, ...]
+end
+
+#! Computes leading number of ones in `number`.
+#!
+#! Stack transition:
+#! Input: [number, ...]
+#! Output: [leading_ones, ...]
+#! Cycles:
+#! - 13 + 11 * leading_ones, when fewer than 32 leading true bits
+#! - 18 + 11 * leading_ones, when more than 32 leading true bits
+export.leading_ones
```

Comment:
Is the name correct here? It seems like we are computing trailing ones rather than leading ones.

Also, I think we should be able to do this more efficiently. Let's create an issue to address this in the future.

Reply:
renamed to trailing_ones
issue created https://github.com/0xPolygonMiden/miden-vm/issues/863